sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatusListener.scala

```diff
@@ -45,13 +50,15 @@ class StreamingQueryStatusListener(sqlConf: SQLConf) extends StreamingQueryListe
 
   override def onQueryStarted(event: StreamingQueryListener.QueryStartedEvent): Unit = {
     activeQueryStatus.putIfAbsent(event.runId,
-      new StreamingQueryUIData(event.name, event.id, event.runId))
+      new StreamingQueryUIData(event.name, event.id, event.runId, event.submitTime))
   }
 
   override def onQueryProgress(event: StreamingQueryListener.QueryProgressEvent): Unit = {
+    val batchTimestamp = timestampFormat.parse(event.progress.timestamp).getTime
     val queryStatus = activeQueryStatus.getOrDefault(
       event.progress.runId,
-      new StreamingQueryUIData(event.progress.name, event.progress.id, event.progress.runId))
+      new StreamingQueryUIData(event.progress.name, event.progress.id, event.progress.runId,
```

Comment:
My intention was not to touch event but be conditional if the UI data is created with onQueryProgress, but the change looks even better to me. I guess adding the field in QueryStartedEvent is OK, as it's marked as Evolving.

Reply:
Thanks for your meticulous and patient review. :)

Reply:
Looks like the change is not incorporated with existing test suites. Could you fix them?

Reply:
Done, but miss a mima failure. Let me fix it.